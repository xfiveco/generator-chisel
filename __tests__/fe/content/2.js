const fs = require('fs-extra');
const { defaultAnswers, md, json } = require('./helpers');

describe('FE Content', () => {
  test('Twig functions work', async () => {
    await global.chiselTestHelpers.generateProjectWithAnswers(
      ['create', '--skip-format-and-build'],
      defaultAnswers,
    );

    fs.outputFileSync('./content/nested.md', md('Nested Top Level'));
    fs.outputFileSync('./content/nested/page.md', md('Nested Page'));
    fs.outputFileSync(
      './content/really/no-parent-nested/page.md',
      md('Nested Page No Parent'),
    );
    fs.outputFileSync('./content/top-page.md', md('Top Page'));
    fs.outputFileSync('./content/json-page.json', json({ title: 'JSON Page' }));
    fs.outputFileSync('./src/assets/hello.txt', 'Hello World');
    fs.outputFileSync('./src/assets/deeply/nested/file.txt', 'DNF');

    fs.outputFileSync(
      './content/class-name.twig',
      `
        <!DOCTYPE html>
        <ul>
          <li>{{ className({}) }}</li>
          <li>{{ className('') }}</li>
          <li>{{ className('c-hello') }}</li>
          <li>{{ className('c-hello', 'world', '', false, null, 'something', none, (isDev() ? 'dev' : 'prod')) }}</li>
        </ul>
      `,
    );

    fs.outputFileSync(
      './content/get-posts.twig',
      /* HTML */ `
        <!DOCTYPE html>
        {% set posts = getPosts({ parent: { '$exists': false } }) %}
        <ul>
          {% for p in posts %}
          <li>
            {{ p.id }} - {{ p.children | length }} <br />
            ({{ p.prev.id}} - {{ p.next.id }})
          </li>
          {% endfor %}
        </ul>
      `,
    );

    fs.outputFileSync(
      './src/templates/post.twig',
      /* HTML */ `
        <!DOCTYPE html>
        <html data-webpack-public-path="{{ getDistPath() }}">
          <head>
            <title>
              {{ post.ID }} - {{ post.type }} - {{ post.parent.id ?: 'no p' }}
            </title>
            <link
              rel="stylesheet"
              property="stylesheet"
              href="{{ revisionedPath('styles/main.css') }}"
              type="text/css"
            />
            <script src="{{ revisionedPath('scripts/app.js') }}" defer></script>
          </head>
          <body>
            <div>
              Here is <a href="{{ assetPath('hello.txt') }}">Text File</a> and
              <a href="{{ assetPath('deeply/nested/file.txt')}}">DNF</a>.
            </div>
            {% set np = getPosts({ id: 'nested/page' })[0] %}
            <div>
              Links:
              <ul>
                <li>Absolute <a href="{{ np.link }}">{{ np.title }}</a></li>
                <li>
                  Relative 1 <a href="{{ np.link(post) }}">{{ np.title }}</a>
                </li>
                <li>
                  Relative 2 <a href="{{ postLink(np) }}">{{ np.title }}</a>
                </li>
              </ul>
            </div>
          </body>
        </html>
      `,
    );

    await global.chiselTestHelpers.runChiselScript(['build']);

    const files = [
      'class-name',
      'get-posts',
      'top-page',
      'json-page',
      'nested/page',
      'really/no-parent-nested/page',
    ];

    for (const file of files) {
      await global.chiselTestHelpers.fileMatchesSnapshot(`./dist/${file}.html`);
    }
  });
});
