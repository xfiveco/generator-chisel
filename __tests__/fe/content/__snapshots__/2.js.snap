// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`FE Content Twig functions work 1`] = `
"<!DOCTYPE html>
<ul>
  <li></li>
  <li></li>
  <li>c-hello</li>
  <li>c-hello c-hello--world c-hello--something c-hello--prod</li>
</ul>"
`;

exports[`FE Content Twig functions work 2`] = `
"<!DOCTYPE html>
<ul>
  <li> class-name - 0 <br /> ( - get-posts) </li>
  <li> get-posts - 0 <br /> (class-name - json-page) </li>
  <li> json-page - 0 <br /> (get-posts - nested) </li>
  <li> nested - 1 <br /> (json-page - nested/page) </li>
  <li> really/no-parent-nested/page - 0 <br /> (nested/page - top-page) </li>
  <li> top-page - 0 <br /> (really/no-parent-nested/page - ) </li>
</ul>"
`;

exports[`FE Content Twig functions work 3`] = `
"<!DOCTYPE html>
<html data-webpack-public-path=\\"./\\">

<head>
  <title> top-page - md - no p </title>
  <link rel=\\"stylesheet\\" property=\\"stylesheet\\" href=\\"styles/main.--HASH--.css\\" type=\\"text/css\\" />
  <script src=\\"scripts/app.98db252c.js\\" defer></script>
</head>

<body>
  <div> Here is <a href=\\"assets/hello.77a781b9.txt\\">Text File</a> and <a href=\\"assets/nested/file.a5d6752d.txt\\">DNF</a>. </div>
  <div> Links: <ul>
      <li>Absolute <a href=\\"/nested/page.html\\">Nested Page</a></li>
      <li> Relative 1 <a href=\\"nested/page.html\\">Nested Page</a>
      </li>
      <li> Relative 2 <a href=\\"nested/page.html\\">Nested Page</a>
      </li>
    </ul>
  </div>
</body>

</html>"
`;

exports[`FE Content Twig functions work 4`] = `
"<!DOCTYPE html>
<html data-webpack-public-path=\\"./\\">

<head>
  <title> json-page - json - no p </title>
  <link rel=\\"stylesheet\\" property=\\"stylesheet\\" href=\\"styles/main.--HASH--.css\\" type=\\"text/css\\" />
  <script src=\\"scripts/app.98db252c.js\\" defer></script>
</head>

<body>
  <div> Here is <a href=\\"assets/hello.77a781b9.txt\\">Text File</a> and <a href=\\"assets/nested/file.a5d6752d.txt\\">DNF</a>. </div>
  <div> Links: <ul>
      <li>Absolute <a href=\\"/nested/page.html\\">Nested Page</a></li>
      <li> Relative 1 <a href=\\"nested/page.html\\">Nested Page</a>
      </li>
      <li> Relative 2 <a href=\\"nested/page.html\\">Nested Page</a>
      </li>
    </ul>
  </div>
</body>

</html>"
`;

exports[`FE Content Twig functions work 5`] = `
"<!DOCTYPE html>
<html data-webpack-public-path=\\"../\\">

<head>
  <title> nested/page - md - nested </title>
  <link rel=\\"stylesheet\\" property=\\"stylesheet\\" href=\\"../styles/main.--HASH--.css\\" type=\\"text/css\\" />
  <script src=\\"../scripts/app.98db252c.js\\" defer></script>
</head>

<body>
  <div> Here is <a href=\\"../assets/hello.77a781b9.txt\\">Text File</a> and <a href=\\"../assets/nested/file.a5d6752d.txt\\">DNF</a>. </div>
  <div> Links: <ul>
      <li>Absolute <a href=\\"/nested/page.html\\">Nested Page</a></li>
      <li> Relative 1 <a href=\\"page.html\\">Nested Page</a>
      </li>
      <li> Relative 2 <a href=\\"page.html\\">Nested Page</a>
      </li>
    </ul>
  </div>
</body>

</html>"
`;

exports[`FE Content Twig functions work 6`] = `
"<!DOCTYPE html>
<html data-webpack-public-path=\\"../../\\">

<head>
  <title> really/no-parent-nested/page - md - no p </title>
  <link rel=\\"stylesheet\\" property=\\"stylesheet\\" href=\\"../../styles/main.--HASH--.css\\" type=\\"text/css\\" />
  <script src=\\"../../scripts/app.98db252c.js\\" defer></script>
</head>

<body>
  <div> Here is <a href=\\"../../assets/hello.77a781b9.txt\\">Text File</a> and <a href=\\"../../assets/nested/file.a5d6752d.txt\\">DNF</a>. </div>
  <div> Links: <ul>
      <li>Absolute <a href=\\"/nested/page.html\\">Nested Page</a></li>
      <li> Relative 1 <a href=\\"../../nested/page.html\\">Nested Page</a>
      </li>
      <li> Relative 2 <a href=\\"../../nested/page.html\\">Nested Page</a>
      </li>
    </ul>
  </div>
</body>

</html>"
`;
